apps/wing/src/analytics/storage.ts

```diff
@@ -0,0 +1,202 @@
+        return undefined;
+      }
+      const eventId = uuidv4();
+      
+      if (!existsSync(this.analyticsStorageDir)) {
+        mkdirSync(this.analyticsStorageDir);
+      }
+  
+      const analyticReportFile = path.join(this.analyticsStorageDir, `${eventId}.json`);
+      
+      
+      // attach timestamp and anonymousId to event
+      event.timestamp = event.timestamp ?? new Date().toISOString();
+      
+      // We add the anonymousId only if we are not in a CI environment
+      if (event.properties.ci) {
+        const anonymousId = this.getAnonymousId();
+        event.anonymousId = anonymousId;
+      }
+
+      this.saveEvent(analyticReportFile, event);
+    
+      return analyticReportFile;
+  
+    } catch (error) {
+      return undefined;
+    }
+  }
+
+  /**
+   * Retrieves an analytic event from disk
+   * 
+   * @param filePath the path the event was saved to
+   * @returns the event or undefined if there was an error
+   */
+  public loadEvent(filePath: string): AnalyticEvent | undefined {
+    try {
+      const fileContents = readFileSync(filePath, 'utf-8');
+      return JSON.parse(fileContents) as AnalyticEvent;
+    } catch (error) {
+      // ignore
+      return undefined;
+    }
+  }
+
+  /**
+   * Reads the analytics config for the user's anonymous id, 
+   * if an id does not exist, one is generated and saved to disk
+   * 
+   * @returns the anonymous id for the user
+   */
+  public getAnonymousId(): string {
+    let config = this.loadConfig();
+    if (!config.anonymousId) {
+      config.anonymousId = this.generateAnonymousId();
+      this.saveConfig(config);
+    }
+    return config.anonymousId;
+  }
+
+
+  private generateAnonymousId(): string {
+    return randomBytes(16).toString('hex');
+  }
+
+  /**
+   * Retrieves the analytics config from disk, if one does not exist
+   * a new one is created and saved to disk
+   * 
+   * @returns the analytics config for the user
+   */
+  public loadConfig(): AnalyticsConfig {
+    try {
+      const fileContents = readFileSync(this.analyticsConfigFile, 'utf-8');
+      return JSON.parse(fileContents) as AnalyticsConfig;
+    } catch (error: any) {
+      // Possible issue reading analytics config file, first determine if file exists yet
+  
+      if (error.code === 'EACCES') {
+        throw new Error("Not able to access analytics config file due to permissions");
+      }
+  
+      const analyticsConfig: AnalyticsConfig = {
+        anonymousId: this.generateAnonymousId(),
+        optOut: false,
+      }
+  
+      this.saveConfig(analyticsConfig);
+  
+      return analyticsConfig;
+    }
+  }
+
+  /**
+   * Saves the analytics config to disk
+   * 
+   * @param config the analytics config to save to disk
+   */
+  public saveConfig(config: AnalyticsConfig) {
+    if (!existsSync(WING_HOME_DIR)) {
+      mkdirSync(WING_HOME_DIR);
+    }
+    writeFileSync(this.analyticsConfigFile, JSON.stringify(config))
+  }
```

Comment:
Maybe you should guard this with try catch ? (we might find this is a readonly fs)